### DISCUSSION TITLE
Don not trim text

### DISCUSSION THREAD

**Discussion by jaldazSTI** [QuestPDF/QuestPDF#103] [Q&A]
Is there a way to stop the trimming process when inserting a Text?
I' looking for something similar to &nbsp; that is used in html.

**Comment by MarcinZiabek** [comment]
Hi! Can you please describe what do you mean by "text trimming"? I am not sure if you are thinking about text wrapping or there is a bug somewhere 😅

**Reply by jaldazSTI** [reply]
stack.Item().Text("   Have spacing at start of line   ");
The expectations is to have the PDF with the blank spaces and the start and end of the line(s).

**Reply by MarcinZiabek** [reply]
Yes, you are right! The current behaviour is inspired by how HTML handles and renders text. Possibly, the trimming process (although the reason is slightly different) can be removed from the beginning and end of the text block. I need to think about it.

Can you please elaborate more about use cases for leaving and therefore showing blank spaces?

**Reply by jaldazSTI** [reply]
I have preformatted documents that need to be render exactly as they where sent.
e.g. Legal documents, medical documents, lab results.

**Comment by DawidRobertNowak** [comment]
Hello @jaldazSTI,

We apologize for the significant delay in getting back to you. Thank you for bringing this up — we understand the importance of preserving exact formatting, especially for preformatted content like legal or medical documents.

This is a great use case. The topic is in our backlog and we are considering creating a dedicated tutorial to cover it.

Should you encounter any further problems, please don't hesitate to open a new issue.

Thank you for your patience!
